src/Scripting/ScriptingSystem.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <string_view>
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <memory>
+#include <utility>
+#include <sol/sol.hpp>
+
+class IBindings;
+
+class ScriptingSystem {
+ public:
+    ScriptingSystem(
+        std::string_view scriptFolder,
+        const std::vector<std::string> &entryPointFiles
+    );
+
+    static std::unique_ptr<ScriptingSystem> create(
+        std::string_view scriptFolder,
+        const std::vector<std::string> &entryPointFiles);
```

Comment:
This is redundant, there's a constructor that does the same. Just use `make_unique` at call site.